### PR TITLE
Jerry mutable records convert

### PR DESCRIPTION
**Pull request by JerryNixon**
## Why make this change?

- Reference associated issue using `#` syntax. e.g. Closes #XX
  - Include summary (1-2 sentences) of linked issue to avoid redirecting reviewers to different pages.
  - Include links to any additional discussion threads related to this change.

## What is this change?

- Summary of how your changes work to give reviewers context of your intent.
  - Links to relevant documentation / StackOverflow, if applicable.

## How was this tested?

- [ ] Integration Tests
- [ ] Unit Tests

## Sample Request(s)

- Example REST and/or GraphQL request to demonstrate modifications
- Example of CLI usage to demonstrate modifications

### CODE REVIEW

**Comment by aaronpowell** [comment]
Is there value in keeping the types as `record` if they are being moved to mutable with writable properties, or should they be made into standard classes?

File: src/Config/ObjectModel/ApplicationInsightsOptions.cs [hunk 1]
```diff
@@ -6,5 +6,36 @@ namespace Azure.DataApiBuilder.Config.ObjectModel;
 /// <summary>
 /// Represents the options for configuring Application Insights.
 /// </summary>
-public record ApplicationInsightsOptions(bool Enabled = false, string? ConnectionString = null)
-{ }
+/// <remarks>This record is mutable.</remarks>
+public record ApplicationInsightsOptions
```
Comment (Aniruddh25) [inline src/Config/ObjectModel/ApplicationInsightsOptions.cs:10]:
```suggestion
public class ApplicationInsightsOptions
```

File: src/Config/ObjectModel/ApplicationInsightsOptions.cs [hunk 2]
```diff
@@ -6,5 +6,36 @@ namespace Azure.DataApiBuilder.Config.ObjectModel;
...
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ApplicationInsightsOptions"/> class.
+    /// </summary>
+    public ApplicationInsightsOptions()
+    {
+        Enabled = default!;
+        ConnectionString = default!;
+    }
```
Comment (seantleonard) [inline src/Config/ObjectModel/ApplicationInsightsOptions.cs]: this seems redundant when the other constructor has default values defined.

**Review by seantleonard — APPROVED** [review]
just needs merge conflict resolved

File: src/Config/ObjectModel/ApplicationInsightsOptions.cs (hunk 1 above)
Comment (JerryNixon) [reply src/Config/ObjectModel/ApplicationInsightsOptions.cs:10]: Mutable records are the plan.

**Pull request merged**